tolstoy/README.md

```diff
@@ -0,0 +1,135 @@
+
+## Overview
+### Very briefly
+Tolstoy will synchronize a local Mentat database against a remote server, modifying local state if necessary, and uploading changes to the server if necessary. Schema additions are allowed (adding vocabulary). Schema mutations are currently not implemented (changing vocabulary). Mentat's core schema must be the same.
+
+**Basic example:**
+
+Client 1 knows about `name` and `age` of a person.
+```
+[
+    {
+        :db/ident :person/name
+        :db/valueType :db.type/string
+        :db/cardinality :db.cardinality/one
+    }
+    {
+        :db/ident :person/age
+        :db/valueType :db.type/long
+        :db/cardinality :db.cardinality/one
+    }
+    {:person/name "Grisha" :person/age 30}
+]
+```
+
+Client 2 doesn't know about `age`, but knows about `ssn`:
+```
+[
+    {
+        :db/ident :person/name
+        :db/valueType :db.type/string
+        :db/cardinality :db.cardinality/one
+    }
+    {
+        :db/ident :person/ssn
+        :db/valueType :db.type/long
+        :db/cardinality :db.cardinality/one
+        :db/unique :db.unique/identity
+        :db/index true
+    }
+    {:person/name "Grisha" :person/ssn 123}
+]
+```
+Sync Client 1, then Client 2, then Client 1 again.
+
+Entity `Grisha` will be "duplicated", since `:person/name` is not defined as unique.
+```
+[
+    [:person/name :db/ident :person/name]
+    [:person/name :db/valueType :db.type/string]
+    [:person/name :db/cardinality :db.cardinality/one]
+    
+    [:person/age :db/ident :person/age]
+    [:person/age :db/valueType :db.type/long]
+    [:person/age :db/cardinality :db.cardinality/one]
+    
+    [:person/ssn :db/ident :person/ssn]
+    [:person/ssn :db/valueType :db.type/long]
+    [:person/ssn :db/cardinality :db.cardinality/one]
+    [:person/ssn :db/unique :db.unique/identity]
+    [:person/ssn :db/index true]
+    
+    [?grisha_one :person/name "Grisha"]
+    [?grisha_one :person/age 30]
+
+    [?grisha_two :person/name "Grisha"]
+    [?grisha_two :person/ssn 123]
+]
+```
+
+If, instead, `:person/name` was defined as `:db/unique :db.unique/identity`, then our final state will be:
+```
+[
+    [...schema datoms...]
+
+    [?united_grisha :person/name "Grisha"]
+    [?united_grisha :person/age 30]
+    [?united_grisha :person/ssn 123]
+]
+```
+
+Note that in above examples, symbols such as `?grisha_one` are meant to expand to some internal entitiy id (e.g. 65536).
+
+### In more detail...
+Syncing is defined in terms of coming to an agreement between local and remote states. A local state is what's currently present on the current instance. A remote state is what's currently present on a server.
+
+We're synchronizing transaction logs are synchronized, and so we may think about the primitive operations in context of smushing together two logs - local and remote.
```

Comment:
nit: this sentence got caught in an edit.

Reply:
Oh man, it did indeed.